### PR TITLE
Fix file.directory clean=true with wildcard requisites

### PR DESCRIPTION
**Pull request by frebib**
States in the following form were not honouring the the requisite states
that manage files when clean=true was applied. This fixes that by
matching filenames using fnmatch, inspired by the original change that
implemented wildcard requisites: a36b3d1858ad127bfe87b73c9faf737d7309bfb1

Signed-off-by: Joe Groocock <jgroocock@cloudflare.com>

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

---

The tests need more work. Ideally this would be implemented as a unit test, but I can't work out where is the correct place to put said tests. Any feedback/advice is welcome

### CODE REVIEW

**Comment by frebib** [comment]
Tests seem to be working and passing locally now

File: tests/pytests/integration/states/test_file.py [hunk 1]
```diff
@@ -480,6 +480,55 @@ def test_issue_60203(
...
+        - clean: true
+        - require:
+          - file: {tmp_path}/*
+
+    {create_file}:
+      file.managed:
+        - contents: foo bar baz
+    """.format(
+            tmp_path=tmp_path,
+            create_file=create_file,
```
Comment (MKLeb) [inline tests/pytests/integration/states/test_file.py:505]:
```suggestion
        tmp_path=tmp_path,
        create_file=create_file,
```

**Comment by MKLeb** [comment]
The added test seems to be failing on windows here.

File: tests/pytests/integration/states/test_file.py (hunk 1 above)
Comment (frebib) [reply tests/pytests/integration/states/test_file.py:505]: The formatting here is what `black` provided. Should I deviate from that? It'll only get reverted

File: tests/pytests/integration/states/test_file.py [hunk 2]
```diff
@@ -480,6 +480,55 @@ def test_issue_60203(
...
+        # Ensure the file was created
+        state_run = next(states)
+        assert state_run["result"] is True
+        assert create_file.is_file()
+
+        # Check to make sure the untracked file was cleaned up
+        state_run = next(states)
+        assert not clean_path.exists()
+        assert "removed" in state_run["changes"]
+        assert str(clean_path) in state_run["changes"]["removed"]
```
Comment (frebib) [inline tests/pytests/integration/states/test_file.py:529]: Windon't
```
[2022-09-19T20:34:44.657Z]        >           assert str(clean_path) in state_run["changes"]["removed"]
[2022-09-19T20:34:44.657Z]        E           AssertionError: assert 'C:\\Users\\Administrator\\AppData\\Local\\Temp\\pytest-of-Administrator\\pytest-0\\test_clean_with_wildcard_requi0\\bar' in ['c:\\users\\administrator\\appdata\\local\\temp\\pytest-of-administrator\\pytest-0\\test_clean_with_wildcard_requi0\\bar']
[2022-09-19T20:34:44.657Z]        E            +  where 'C:\\Users\\Administrator\\AppData\\Local\\Temp\\pytest-of-Administrator\\pytest-0\\test_clean_with_wildcard_requi0\\bar' = str(WindowsPath('C:/Users/Administrator/AppData/Local/Temp/pytest-of-Administrator/pytest-0/test_clean_with_wildcard_requi0/bar'))
```

File: tests/pytests/integration/states/test_file.py (hunk 1 above)
Comment (Ch3LL) [reply tests/pytests/integration/states/test_file.py:505]: I would stick to black or else the tests will fail. thanks

File: tests/pytests/integration/states/test_file.py (hunk 2 above)
Comment (MKLeb) [reply tests/pytests/integration/states/test_file.py:529]: It seems the state_run path is all lowercase, so maybe converting the LHS to lower will make it pass?

File: tests/pytests/integration/states/test_file.py (hunk 1 above)
Comment (MKLeb) [reply tests/pytests/integration/states/test_file.py:505]: It looks as though our pre-commit check is running black and failing because of these two lines

**Comment by Ch3LL** [comment]
@frebib any luck with @MKLeb 's suggestion?

**Comment by frebib** [comment]
This slipped through the cracks and I'm on PTO for another week. I'll try to remember to look at this when I'm back

**Review by Ch3LL — APPROVED** [review]

**Comment by dwoz** [comment]
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

**Pull request closed without merging**